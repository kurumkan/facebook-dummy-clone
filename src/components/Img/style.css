.image-wrapper {
    position: relative;
    display: inline-block;
    vertical-align: top;
    overflow: hidden;
}

.image-wrapper img {
    max-width: 100%;
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
}

.image-wrapper .image {
    opacity: 0;
    transition: all 1s;
}

.image-wrapper .loaded {
    opacity: 1;
}

.image-wrapper .placeholder {
    transition: all 1s;
}

.image-wrapper .placeholder.hidden {
    display: none;
}

