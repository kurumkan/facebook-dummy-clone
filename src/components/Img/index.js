import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './style.css';

class Img extends Component {
  constructor(props) {
    super(props);
    this.state = {
      imageLoaded: false
    };
  }
  handleLoad = () => {
    this.setState({
      imageLoaded: true
    });
  };

  render() {
    const { imageUrl, width, placeholder } = this.props;
    if(!imageUrl) {
      return <img src={placeholder} />;
    }
    return (
      <div className="image-wrapper" style={{ width, paddingTop: this.state.imageLoaded ? `${this.image.height}px` : width }}>
        <img
          src={placeholder}
          className={`placeholder ${this.state.imageLoaded && 'hidden'}`}
        />
        <img
          src={ imageUrl }
          onLoad={this.handleLoad}
          className={`image ${this.state.imageLoaded && 'loaded'}`}
          ref={el => this.image = el}
        />
      </div>
    );
  }
}

Img.propTypes = {
  imageUrl: PropTypes.string,
  width: PropTypes.string,
  placeholder: PropTypes.string
};

Img.defaultProps = {
  imageUrl: '',
  width: '100%',
  placeholder: ''
};

export default Img;