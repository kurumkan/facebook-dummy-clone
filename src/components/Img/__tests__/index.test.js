import Img from '../index.js';
import React from 'react';
import renderer from 'react-test-renderer';
import ImgPlaceholder from 'images/image-placeholder.jpg';
import ImgToLoad from 'images/image-to-load.jpg';

describe('Img', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(
      <Img
        imageUrl={ImgToLoad}
        width="100%"
        placeholder={ImgPlaceholder}
      />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});