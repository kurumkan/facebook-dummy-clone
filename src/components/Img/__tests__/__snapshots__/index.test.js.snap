// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Img renders correctly 1`] = `
<div
  className="image-wrapper"
  style={
    Object {
      "paddingTop": "100%",
      "width": "100%",
    }
  }
>
  <img
    className="placeholder false"
    src="test-file-stub"
  />
  <img
    className="image false"
    onLoad={[Function]}
    src="test-file-stub"
  />
</div>
`;
